src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationScheduleTest.kt

```diff
@@ -0,0 +1,107 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity
+
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.PENTONVILLE
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.isBool
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.probationTeam
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.UUID
+
+class LocationScheduleTest {
+  private val saturday = LocalDate.of(2025, 2, 22).atStartOfDay()
+  private val sunday = LocalDate.of(2025, 2, 23).atStartOfDay()
+  private val mondayNoon = LocalDate.of(2025, 2, 24).atTime(12, 0)
+  private val mondayMidnight = LocalDate.of(2025, 2, 24).atTime(0, 0)
+
+  @DisplayName("Probation tests")
+  @Nested
+  inner class Probation {
+    @Test
+    fun `should be available for probation team on Monday at noon`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION, allowedParties = "PROBATION_TEAM")
+
+      schedule.isAvailableForProbationTeam(probationTeam(code = "PROBATION_TEAM"), mondayNoon) isBool true
+    }
+
+    @Test
+    fun `should be available for probation team on Monday out of hours`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION, allowedParties = "PROBATION_TEAM")
+
+      schedule.isAvailableForProbationTeam(probationTeam(code = "PROBATION_TEAM"), mondayMidnight) isBool true
+    }
+
+    @Test
+    fun `should be not available for probation team on Monday at noon`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION, allowedParties = "PROBATION_TEAM")
+
+      schedule.isAvailableForProbationTeam(probationTeam(code = "DIFFERENT_PROBATION_TEAM"), mondayNoon) isBool false
+    }
+
+    @Test
+    fun `should be available for any probation team on Monday at noon`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION)
+
+      schedule.isAvailableForAnyProbationTeam(mondayNoon) isBool true
+    }
+
+    @Test
+    fun `should be available for any probation team on Monday out of hours`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION)
+
+      schedule.isAvailableForAnyProbationTeam(mondayMidnight) isBool true
+    }
+
+    @Test
+    fun `should not be available for any probation team on Monday at noon`() {
+      val schedule = schedule(locationUsage = LocationUsage.PROBATION, allowedParties = "PROBATION_TEAM")
+
+      schedule.isAvailableForAnyProbationTeam(mondayNoon) isBool false
+    }
```

Comment:
Good set of tests. I'm sure there will be other variations related to multiple schedules in place, any one of which might allow? But probably a higher level test.

Reply:
This is s tricky one.  The number of permutations is high, so the balance is a fine line.